Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -95,16 +113,45 @@ class BTAnalyticsService: Equatable {
             if let http = self.http, http.baseURL.absoluteString == "test://do-not-send.url" {
                 return
             }
+            
+            if shouldBypassTimerQueue {
+                await self.sendQueuedAnalyticsEvents(configuration: configuration)
+                return
+            }
+
+            if BTAnalyticsService.timer == nil {
+                BTAnalyticsService.timer = DispatchSource.makeTimerSource(queue: self.http?.dispatchQueue)
+                BTAnalyticsService.timer?.schedule(
+                    deadline: .now() + .seconds(self.timerInterval),
+                    repeating: .seconds(self.timerInterval),
+                    leeway: .seconds(1)
+                )
 
-            let postParameters = self.createAnalyticsEvent(config: configuration, sessionID: self.apiClient.metadata.sessionID, event: event)
-            self.http?.post("v1/tracking/batch/events", parameters: postParameters) { _, _, _ in }
+                BTAnalyticsService.timer?.setEventHandler {
```

Comment:
Another question I had was right now in main, individual analytics event is sent to v1/tracking/batch/events which handles batching of the events on the server side?

Reply:
We were using `v1/tracking/batch/events` to send individual events not batched events. That API does not handle batching at all, just allows you to send events optionally in batches. This PR starts sending events in batches when available, though sometimes based on the timer we may still send a singular event to this endpoint in some cases.

Reply:
Another question I had was, I am not familiar with this timer, what happens if timer gets fired while events are in flight, being sent to back end.

Reply:
It would send whatever the current array of events is and any subsequent events would be appended to the now empty array of events.